cmd/juju/romulus/allocate/allocate.go

```diff
@@ -1,100 +1,88 @@
-	Limit     string
+	api   apiClient
+	Value string
 }
 
-// NewAllocateCommand returns a new allocateCommand
-func NewAllocateCommand() modelcmd.ModelCommand {
-	return &allocateCommand{}
+// NewUpdateAllocationCommand returns a new updateAllocationCommand.
+func NewUpdateAllocationCommand() modelcmd.ModelCommand {
+	return &updateAllocationCommand{}
 }
 
-const doc = `
-Allocate budget for the specified applications, replacing any prior allocations
-made for the specified applications.
+func (c *updateAllocationCommand) newAPIClient(bakery *httpbakery.Client) (apiClient, error) {
+	if c.api != nil {
+		return c.api, nil
+	}
+	c.api = api.NewClient(bakery)
+	return c.api, nil
+}
 
-Examples:
-    # Assigns application "db" to an allocation on budget "somebudget" with
-    # the limit "42".
-    juju allocate somebudget:42 db
+type apiClient interface {
+	UpdateAllocation(string, string) (string, error)
+}
 
-    # Application names assume the current selected model, unless otherwise
-    # specified with:
-    juju allocate -m [<controller name:]<model name> ...
+const doc = `
+Updates an existing allocation for a model.
 
-    # Models may also be referenced by UUID when necessary:
-     juju allocate --model-uuid <uuid> ...
+Examples:
+    # Sets the allocation for the current model to 10.
+    juju update-allocation 10
 `
 
-// SetFlags implements cmd.Command.SetFlags.
-func (c *allocateCommand) SetFlags(f *gnuflag.FlagSet) {
-	c.ModelCommandBase.SetFlags(f)
-	f.StringVar(&c.ModelUUID, "model-uuid", "", "Model UUID of allocation")
-}
-
 // Info implements cmd.Command.Info.
-func (c *allocateCommand) Info() *cmd.Info {
+func (c *updateAllocationCommand) Info() *cmd.Info {
 	return &cmd.Info{
 		Name:    "allocate",
-		Args:    "<budget>:<value> <application> [<application2> ...]",
-		Purpose: "Allocate budget to applications.",
+		Args:    "<value>",
+		Purpose: "Update an allocation.",
 		Doc:     doc,
 	}
 }
 
 // Init implements cmd.Command.Init.
-func (c *allocateCommand) Init(args []string) error {
-	if len(args) < 2 {
-		return errors.New("budget and application name required")
+func (c *updateAllocationCommand) Init(args []string) error {
+	if len(args) < 1 {
+		return errors.New("value required")
 	}
-	budgetWithLimit := args[0]
-	var err error
-	c.Budget, c.Limit, err = parseBudgetWithLimit(budgetWithLimit)
-	if err != nil {
-		return errors.Annotate(err, `expected args in the form "budget:limit [application ...]"`)
-	}
-	if c.ModelUUID == "" {
-		c.ModelUUID, err = c.modelUUID()
-		if err != nil {
-			return err
-		}
-	} else {
-		if !utils.IsValidUUIDString(c.ModelUUID) {
-			return errors.NotValidf("model UUID %q", c.ModelUUID)
-		}
+	c.Value = args[0]
+	if _, err := strconv.ParseInt(c.Value, 10, 32); err != nil {
+		return errors.New("value needs to be a whole number")
 	}
+	return cmd.CheckEmpty(args[1:])
+}
 
-	c.Services = args[1:]
-	return nil
+func (c *updateAllocationCommand) modelUUID() (string, error) {
```

Comment:
this would be nice as a method on ModelCommandBase
i wonder if there's a helper there already?